### PR TITLE
celery: redis scheduler backend

### PR DESCRIPTION
**Pull request by rerowep**
## Why are you opening this PR?

- so we can change dynamically scheduled tasks

## How to test?

You have to start a `celery worker beat`. For using this backend we have to call `celery` with following parameter:
`-S rero_ils.schedulers.RedisScheduler`
Example (This is normally in the `server` startup script):
`celery worker -A invenio_app.celery --beat -S rero_ils.schedulers.RedisScheduler`

- run bootstrap
- run setup
- try to start stop scheduled tasks with `invenio scheduler --help` 

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -95,3 +95,9 @@ recursive-include tests *.png
 # flask-wiki to be removed
 recursive-include wiki *.md
 recursive-include wiki *.png
+
+# error files from documents creation
+data/*_errors.xml
```
Comment (jma) [inline MANIFEST.in:100]: exclude?

File: rero_ils/schedulers.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        https://github.com/liuliqiang/redisbeat
+    celery BeaseSchedule:
+        https://github.com/celery/celery/blob/master/celery/schedules.py
+
+    entries: https://docs.celeryproject.org/en/latest/userguide/
+        periodic-tasks.html#entries
+    """
+
+    def __init__(self, app, *args, **kwargs):
+        """Redis scheduler class initializer.
```
Comment (jma) [inline rero_ils/schedulers.py:60]: can you describe the params?

File: rero_ils/schedulers.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+        logger.info('Connect: {url} lazy:{lazy}'.format(
+            url=url,
+            lazy=lazy
+        ))
+        kwargs['app'] = app
+        kwargs['lazy'] = lazy
+        super(RedisScheduler, self).__init__(*args, **kwargs)
+
+    def enabled_name(self, name):
+        """Name for enabled value in REDIS DB."""
```
Comment (jma) [inline rero_ils/schedulers.py:77]: can you describe the params?

File: rero_ils/schedulers.py [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+        """Init entries from CELERY_BEAT_SCHEDULE."""
+        config = deepcopy(self.app.conf.CELERY_BEAT_SCHEDULE)
+        self.merge_inplace(config)
+        msg = 'Current schedule:\n{tasks}'.format(
+            tasks='\n'.join(self.display_all(prefix='- Tasks: '))
+        )
+        logger.info(msg)
+
+    def is_due(self, entry):
+        """Return tuple of ``(is_due, next_time_to_check)``.
```
Comment (jma) [inline rero_ils/schedulers.py:100]: can you describe the params?

File: MANIFEST.in (hunk 1 above)
Comment (rerowep) [reply MANIFEST.in:100]: Yes, because there more or less temporary.

File: rero_ils/schedulers.py (hunk 2 above)
Comment (rerowep) [reply rero_ils/schedulers.py:60]: Oh, there a lot in the kwargs.

**Comment by BadrAly** [comment]
Can we add the instructions above to the developer resources?

**Pull request merged**